lib/modules/datasource/github-releases/cache/cache-base.ts

```diff
@@ -115,6 +115,8 @@ export abstract class AbstractGithubDatasourceCache<
 
   private resetDeltaMinutes: number;
 
+  private cachedPromises: Record<string, Promise<StoredItem[]>> = {};
```

Comment:
Does this get reset after each repo?

Reply:
I believe we re-initialize new datasource instances on each run, so the cache instances will be reset too.

Reply:
Which function does that? I might need to add it to the hosted app

Reply:
The only way I see to avoid long-term cache read race condition is to return the same promise object from function, so two sequential calls will return literally the same object.

Reply:
Sorry, misread your question...

Reply:
I was wrong, datasource instances aren't reinitialized

Reply:
Need to use memory cache instead